Add support for customerId in clientToken request

kpgarrod opened the pull request:
I am trying to use the library to create subscriptions. I ran across a stumbling block which is that I need a customerId in the server to create a clientToken for a subscription. I have just added a `customerId` parameter to the braintree-dropin directive and passed that through to the server as a query parameter.

In the process, I noticed that there was a test for a similar feature and some (apparently partially implemented) code. 

Please feel free to scrap this pull request if you are implementing something more comprehensive :), but this seems to be working for me.

Sorry I didn't update the tests because I couldn't get them to run on my system and didn't have time to fight with them

jeffcarp commented on package.json:
```diff
@@ -23,7 +23,10 @@
+    "braintree-web": "2.7.2",
+    "dnode-protocol": "^0.2.2",
+    "mime-db": "^1.20.0",
+    "sockjs": "^0.3.15"
```
It doesn't look like these are used in the PR?

jeffcarp commented:
Hey @kpgarrod, thanks for the PR :smile: 

There has been some work toward this but it has kind of stalled. It'd be good to get this functionality into master soon. I left a small comment on the PR but other than that it looks good overall.

kpgarrod commented:
@jeffcarp I had to add those dependencies to get the 'test' script to build the project. `npm build` didn't seem to do anything at all and `npm test` failed a couple of times until I added those dependencies. Once I had added them, `npm test` got as far as building the project and then the tests failed. I didn't take it any further. I wasn't sure if it was something specific to my system or not. BTW, it's off-topic, but the Braintree form takes a while to load and it leaves my page sitting there looking a bit silly. Is there a way to know when the form has loaded? Thanks for the library!

jeffcarp commented:
Not sure if this will help, but I think it's `npm run build` instead of `npm build`. That's odd, do you have steps to reproduce? It doesn't look like Travis CI has a problem running the tests from a fresh start.

Re: knowing when BT has loaded - there's an `onReady` callback event you can hook into ([docs](https://developers.braintreepayments.com/guides/client-sdk/javascript/v2#global-setup)). Let me know if that helps.

jeffcarp commented:
Closing due to inactivity
